.githooks/pre-commit.d/checkstyle

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# https://stackoverflow.com/a/3466183/1675299
+
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    Linux*)     machine=Linux;;
+    Darwin*)    machine=Mac;;
+    CYGWIN*)    machine=Cygwin;;
+    MINGW*)     machine=MinGw;;
+    *)          machine="UNKNOWN:${unameOut}"
+esac
+
+if [[ "$machine" == "Cygwin" || "$machine" == "MinGw" ]]; then
+    ./gradlew.bat checkstyleMain
```

Comment:
Perhaps we could run `checkstyleTest` as well by doing `./gradlew.bat check coverage`. same for line 18 below

Reply:
I think only running `check` is needed